examples/model_selection/plot_grid_search_refit_callable.py

```diff
@@ -0,0 +1,125 @@
+"""
+=======================================================================
+Balance model complexity and cross-validated score using refit=callable
+=======================================================================
+
+A simple example demonstrates the usage of `refit=callable` interface in 
+`sklearn.model_selection.GridSearchCV`. It shows this interface adds certain
+amount of flexibility in identifying the "best" estimator. The function 
+passed to paramter `refit` incorporate of which metric(s) to optimise. This
+interface can also be used in multiple metrics evaluation.
+
+This example balances model complexity and cross-validated score by 
+finding a decent accuracy within 1 standard deviation of the best accuracy
+score while minimising the number of PCA components.
+
+The figure shows the trade of between cross-validated score and number of
+PCA components. The balanced case is when n_components=6 and accuracy=0.77,
+which falls into the range within 1 standard deviation of the best accuracy
+score.
+"""
+# Author: Wenhao Zhang <wenhaoz@ucla.edu>
+
+print(__doc__)
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.datasets import load_digits
+from sklearn.decomposition import PCA
+from sklearn.model_selection import GridSearchCV
+from sklearn.pipeline import Pipeline
+from sklearn.svm import LinearSVC
+
+
+def score_bounds(scores):
+    """
+    Calculate the upper/lower bounds within 1 standard deviation
+    of the best `mean_test_scores`.
+
+    Args:
+        scores: 'mean_test_score' here
+
+    Return:
+        upper/lower bounds within 1 standard deviation of the
+        best `mean_test_scores`.
+    """
+    std_test_score = np.std(scores)
```

Comment:
Should be using std_test_score: you want standard deviation across cv splits, not across parameter candidates